lib/checkPullRequestJob.js

```diff
@@ -0,0 +1,266 @@
+const { SERVER_JOBS_ADMIN } = require('../userWhitelist.json');
+const REGISTRY_FILENAME = 'jobs_registry.py';
+const JOB_FILE_FIX = '_jobs_';
+const CRITICAL_LABEL = 'critical';
+
+/**
+ * @param {import('probot').Octokit.PullsListFilesResponseItem}
+ */
+const isJobFile = ({ filename }) => {
+  return filename.includes(JOB_FILE_FIX);
+};
+
+/**
+ * @param {import('probot').Octokit.PullsListFilesResponseItem}
+ */
+const isNewFile = ({ status }) => {
+  return status === 'added';
+};
+
+/**
+ * @param {import('probot').Octokit.PullsListFilesResponseItem} file
+ */
+const getNewJobsFromFile = (file) => {
+  // This represents an new line added from the git diff.
+  // This is done because classes can only be created in a new line.
+  // '+' represents an addition while '-' represents a deletion.
+  const newLineAdded = '\n+';
+  const changesArray = file.patch.split(newLineAdded);
+  const classIdentifier = 'class';
+  const oneOffJobIdentifier = 'OneOffJob';
+
+  // A new job class should start with class and include OneOffJob in its name.
+  const newJobDefinitions = changesArray.filter(
+    (change) =>
+      change.startsWith(classIdentifier) && change.includes(oneOffJobIdentifier)
+  );
+
+  // New job definitions should look like:
+  // class OppiabotContributionsOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+  // To get the actual job name, we need to remove the unneeded items.
+
+  const newJobs = newJobDefinitions.map((definition) => {
+    return definition.substring(
+      // The +1 is to account for the space in the class definition.
+      classIdentifier.length + 1,
+      definition.indexOf('(')
+    );
+  });
+
+  return newJobs;
+};
+
+/**
+ * @param {import('probot').Octokit.PullsListFilesResponseItem} file
+ */
+const addsNewJob = (file) => {
+  const newJobs = getNewJobsFromFile(file);
+  return newJobs.length > 0;
+};
+
+/**
+ * Returns the name of the job from it's file name.
+ * @param {string} filename
+ *
+ * @example
+ * getJobNameFromFileName('core/domain/user_jobs_one_off.py')
+ * // returns user_jobs_one_off
+ */
+const getJobNameFromFileName = (filename) => {
+  // Filename is the relative path to the file, we need to
+  // get the exact file name.
+  const substrings = filename.split('/');
+  const actualFileNameWithExtension = substrings[substrings.length - 1];
+  const extensionIndex = actualFileNameWithExtension.indexOf('.py');
+
+  return actualFileNameWithExtension.substring(0, extensionIndex);
+};
+
+/**
+ * @param {string} job
+ * @param {import('probot').Octokit.PullsListFilesResponse} files
+ */
+const doesRegistryContainFile = (job, files) => {
```

Comment:
Rename to `isRegistryUpdated` since it is checking two things whether registry is modified and if modified does it include the new job.

Reply:
Done.